Bks.DataAccess.Mongo.Test/MongoConnectorTest.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using FakeItEasy;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MongoDB.Driver;
+
+namespace Bks.DataAccess.Mongo.Test
+{
+    [TestClass]
+    public class MongoConnectorTest
+    {
+        private const string DbName = "MY_DB_NAME";
+        private const string DbPrefix = "MY_DB_PREFIX";
+        private const string DbCollectionName = "MY_COLLECTION_NAME";
+
+        private readonly ILogger<MongoConnector> mockLogger = 
+            A.Fake<ILogger<MongoConnector>>();
+        
+        private readonly IReadOnlyCollection<IMongoClassMapper> classMaps =
+            A.Fake<IReadOnlyCollection<IMongoClassMapper>>();
+
+        private static IOptions<MongoSettings> CreateMongoSettingsOptions()
+        {
+            var mongoSettings = new MongoSettings()
+            {
+                ConnectionString = "mongodb://localhost:27017?maxPoolSize=5",
+                Database = DbName,
+                CollectionPrefix = DbPrefix
+            };
+            var mongoSettingsOptions = Options.Create(mongoSettings);
+            return mongoSettingsOptions;
+        }
+
+        [TestMethod]
+        public void RegisterClassMaps_with3Mappers_allMappersAreUsed()
+        {
+            var mapperA = A.Fake<IMongoClassMapper>();
+            var mapperB = A.Fake<IMongoClassMapper>();
+            var mapperC = A.Fake<IMongoClassMapper>();
+            IReadOnlyCollection<IMongoClassMapper> mockedClassMaps = new List<IMongoClassMapper>(){ mapperA , mapperB, mapperC };
+
+            var options = CreateMongoSettingsOptions();
+            var mockMongo = A.Fake<MongoConnector>(
```

Comment:
What do you test here if you mock the MongoConnector?

Reply:
Testing RegisterClassMaps is called and perform the map for each of the class maps used.
I had to use a mock as MongoConnector is an abstract class, but the RegisterClassMaps is still called during the test on each ClassMap